pkg/resource/training_job/sdk.go

```diff
@@ -1087,9 +1113,149 @@ func (rm *resourceManager) sdkUpdate(
 	desired *resource,
 	latest *resource,
 	delta *ackcompare.Delta,
-) (*resource, error) {
-	// TODO(jaypipes): Figure this out...
-	return nil, ackerr.NotImplemented
+) (updated *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.sdkUpdate")
+	defer func() {
+		exit(err)
+	}()
+	input, err := rm.newUpdateRequestPayload(ctx, desired)
+	if err != nil {
+		return nil, err
+	}
+	warmpool_diff := delta.DifferentAt("Spec.ResourceConfig.KeepAlivePeriodInSeconds")
+	profiler_diff := delta.DifferentAt("Spec.ProfilerConfig") || delta.DifferentAt("Spec.ProfilerRuleConfigurations")
+	if warmpool_diff && profiler_diff {
+		return latest, ackerr.NewTerminalError(errors.New("cannot update Warm pool and Profiler at the same time"))
+	}
+	if !warmpool_diff && !profiler_diff {
+		return latest, ackerr.NewTerminalError(errors.New("only Warm Pool or Profiler can be updated"))
+	}
+	if warmpool_diff {
+		input.SetProfilerConfig(nil)
+		input.SetProfilerRuleConfigurations(nil)
+		warmpool_terminal := warmPoolTerminalCheck(latest)
+		if warmpool_terminal {
+			return latest, ackerr.NewTerminalError(errors.New("warm pool either does not exist or has reached a non updatable state"))
+		}
+		//Requeue if TrainingJob is in InProgress state
+		if err := customSetOutputUpdateWarmpool(latest); err != nil {
+			return nil, err
+		}
+	}
+	if profiler_diff {
+		if up_err := customSetOutputUpdateProfiler(latest); up_err != nil {
+			return nil, up_err
+		}
+		input.SetResourceConfig(nil)
+		if profilerRemovalCheck(desired, latest) {
+			handleProfilerRemoval(input)
+		} else {
+			inp_err := customSetUpdateInput(desired, latest, delta, input)
+			if inp_err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	var resp *svcsdk.UpdateTrainingJobOutput
+	_ = resp
+	resp, err = rm.sdkapi.UpdateTrainingJobWithContext(ctx, input)
+	rm.metrics.RecordAPICall("UPDATE", "UpdateTrainingJob", err)
+	if err != nil {
+		return nil, err
+	}
+	// Merge in the information we read from the API call above to the copy of
+	// the original Kubernetes object we passed to the function
+	ko := desired.ko.DeepCopy()
+
+	if ko.Status.ACKResourceMetadata == nil {
+		ko.Status.ACKResourceMetadata = &ackv1alpha1.ResourceMetadata{}
+	}
+	if resp.TrainingJobArn != nil {
+		arn := ackv1alpha1.AWSResourceName(*resp.TrainingJobArn)
+		ko.Status.ACKResourceMetadata.ARN = &arn
+	}
+
+	rm.setStatusDefaults(ko)
+	customSetOutputPostUpdate(ko, delta)
```

Comment:
call a readOne instead of this custom checks and set status from it. See the sdkdelete method for reference